src/api/docroot.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// DO NOT SUBMIT: is DocRoot concept enough? Or do we also need a separate
+// concept of Scope?
+
+/**
+ * `DocRoot` is very similar to the `Document` interface and acts as a stand-in
+ * for the `Document` in most of cases. It's a pseudo-document and a unit of
+ * scoping in AMP. Each independent sub-document will have its own instance of
+ * `DocRoot` even if multiple `DocRoot`s share the same global `Document`
+ * instance.
```

Comment:
Going to be annoying and bring this up again... :) 

Can we evolve and rename `AmpDoc` instead? There's fair amount of overlap between `AmpDoc` and `DocRoot` APIs, and I think AMP engs are already confused about when to use `AmpDoc` vs. `Document`. Needing to grok `AmpDoc` vs. `Document` vs. `DocRoot` only makes this more confusing, especially when APIs are very similar.

Reply:
I want to completely remove `ampdoc` in a "short" order. They are similar, but key API differences and also problem with FIE that will take few PRs to address.

Reply:
I.e. to me this is a migration tool, not the end state.